src/Cache/Cache.php

```diff
@@ -117,23 +119,34 @@ public static function fromJson(string $json): self
 
         $ruleSet = new RuleSet();
         if (\is_array($data['sniffs'])) {
+            /** @psalm-var class-string $sniffName */
             foreach ($data['sniffs'] as $sniffName) {
-                $sniff = new $sniffName();
-                if ($sniff instanceof SniffInterface) {
-                    $ruleSet->addSniff($sniff);
+                try {
```

Comment:
I'll take a look this We if we can improve this.

Reply:
I've committed something better as suggested by someone at Psalm.

Reply:
This is really better indeed. thanks.